numpy/lib/nanfunctions.py

```diff
@@ -1437,3 +1438,245 @@ def nanstd(a, axis=None, dtype=None, out=None, ddof=0, keepdims=np._NoValue):
+
+    >>> x = [-2.1, -1,  np.nan, 4.3]
+    >>> y = [3,  1.1,  1, 0.12]
+    >>> X = np.vstack((x,y))
+    >>> print np.nancov(X)
+    [[ 11.71        -4.286     ]
+     [ -4.286        2.14413333]]
+    >>> print np.nancov(x, y)
+    [[ 11.71        -4.286     ]
+     [ -4.286        2.14413333]]
+    >>> print np.nancov(x)
+    11.71
+
+    Finally it can be seen how the pairwise option handles nan values
+    differently than when when the pairwise option is disabled (default).
+
+    >>> X = np.array([[ 1.,  2.,     10., np.nan,   5.],
+    ...               [ 4., 10.,     13., np.nan,   8.],
+    ...               [-2., -4., np.nan, np.nan, -10.]])
+    >>> np.nancov(X)
+    array([[  4.33333333,   2.66666667,  -8.66666667],
+           [  2.66666667,   9.33333333,  -5.33333333],
+           [ -8.66666667,  -5.33333333,  17.33333333]])
+    >>> np.nancov(X, pairwise=True)
+    array([[ 16.33333333,  12.16666667,  -8.66666667],
+           [ 12.16666667,  14.25      ,  -5.33333333],
+           [ -8.66666667,  -5.33333333,  17.33333333]])
+    """
+    # Check inputs
+    if ddof is not None and ddof != int(ddof):
+        raise ValueError(
+            "ddof must be integer")
+
+    m = np.asarray(m)
+    if m.ndim > 2:
+        raise ValueError("m has more than 2 dimensions")
+
+    if y is None:
+        dtype = np.result_type(m, np.float64)
+    else:
+        y = np.asarray(y)
+        if y.ndim > 2:
+            raise ValueError("y has more than 2 dimensions")
+        dtype = np.result_type(m, y, np.float64)
+    X = np.array(m, ndmin=2, dtype=dtype)
+
+    if X.shape[0] == 1:
+        rowvar = 1
+    if rowvar:
+        axis = 0
+    else:
+        axis = 1
+
+    if y is not None:
+        y = np.array(y, copy=False, ndmin=2, dtype=dtype)
+        X = np.concatenate((X, y), axis)
+
+    if axis:
+        n = X.shape[0]
+    else:
+        n = X.shape[1]
+
+    # Get the product of frequencies and weights
+    if fweights is not None:
+        fweights = np.asarray(fweights, dtype=np.float64)
+        if not np.all(fweights == np.around(fweights)):
+            raise TypeError(
+                "fweights must be integer")
+        if fweights.ndim > 1:
+            raise RuntimeError(
+                "cannot handle multidimensional fweights")
+        if fweights.shape[0] != n:
+            raise RuntimeError(
+                "incompatible numbers of samples and fweights")
+        if any(fweights < 0):
+            raise ValueError(
+                "fweights cannot be negative")
+    if aweights is not None:
+        aweights = np.asarray(aweights, dtype=np.float64)
+        if aweights.ndim > 1:
+            raise RuntimeError(
+                "cannot handle multidimensional aweights")
+        if aweights.shape[0] != n:
+            raise RuntimeError(
+                "incompatible numbers of samples and aweights")
+        if any(aweights < 0):
+            raise ValueError(
+                "aweights cannot be negative")
```

Comment:
I'd be a lot happier if all this could be extracted into a helper function shared by `cov` and `nancov`